core/src/packet_deserializer.rs

```diff
@@ -42,9 +52,15 @@ impl PacketDeserializer {
         capacity: usize,
     ) -> Result<ReceivePacketResults, RecvTimeoutError> {
         let (packet_count, packet_batches) = self.receive_until(recv_timeout, capacity)?;
+
+        // get current root bank to get feature gate status
+        let _root_bank = self.bank_forks.read().unwrap().root_bank();
```

Comment:
I'm not familiar with this specific feature `round_compute_unit_price_enabled`, but should we be getting it from the root bank? It seems like we should be checking the feature gates of the bank we're currently building, or our parent? 
If it's a feature gate whole cluster won't cross epoch on their root bank at the same time which would lead to different timing for this feature to be activated across the cluster.

Reply:
Excellent point! So this PR is prerequisite for https://github.com/solana-labs/solana/pull/31469, where the feature gate and its functions are implemented. 

The context is: the feature gate is required (for #31469) between banking stage receives packet from sigverify, and before inserting it to priority-sorted queue (Transaction storage), because the prioritization (eg compute-unit-price) defined in compute-budget IX will be rounded down to nearest lamports when the feature is activated. 

This PR is getting banking stage part of code ready for that. 

Was thinking about picking one between `root_bank` over `working_bank`, sounds like `working_bank` is the better option for the purpose.

Reply:
94fefbbbf6345931d61a8923ffcbd685b4ca4030

Reply:
Okay makes sense. Since it only affects block packing seems like it wouldn't actually affect consensus? Basically it's okay if some members of the cluster are rounding down and some are not - so does it require a feature gate?

Reply:
This PR only affects leader code; to be more precious, would only affect at what numerical value of priority the packets are sorted in queue. For example, if a transaction sets compute-unit-price to 1_000_900; Before feature is enabled, the transaction will be prioritized as `1_000_900`, after feature activation, it would be prioritized as `1_000_000`. But all this change are in leader's thread queue only. 

Just to clarify, the PR does not change code behavior, it only does the plumbing work. (it hardcoded the flag as `false` as feature is not activated, which is current behavior. )

It is a prerequisite of #31469 , where feature is actually added, and its status will affect how transactions are charged with priority fee, hence could affect consensus.